membership_variable_period/models/product_template.py

```diff
@@ -19,17 +19,17 @@ def _get_next_date(self, date):
         @rtype: date
         @return: The date incremented in 'interval' units of 'unit'.
         """
-        self.ensure_one()
-        if isinstance(date, str):
-            date = fields.Date.from_string(date)
-        if self.membership_interval_unit == 'days':
-            return date + timedelta(days=self.membership_interval_qty)
-        elif self.membership_interval_unit == 'weeks':
-            return date + timedelta(weeks=self.membership_interval_qty)
-        elif self.membership_interval_unit == 'months':
-            return date + relativedelta(months=self.membership_interval_qty)
-        elif self.membership_interval_unit == 'years':
-            return date + relativedelta(years=self.membership_interval_qty)
+        for rec in self:
```

Comment:
This can't be multi, as it returns one value. Leave the `self.ensure_one()` clause, and call it record by record in case of multi-recordset